### ISSUE TITLE
Plugin throws 500 while importing data

### ISSUE THREAD

**Issue by subratom** [engram-design/FeedMe#60]
I have had this issue when I run the plugin from CP and now I face the same issue while using the curl method. The issue is, I have no idea why http 500 error is being thrown. Based on the last entry in contents table I know what was imported but because the task fails I have to empty out all the information before I can start again (No idea on how to restart the task, all efforts to retry have failed). 

Any way to know where I can trap this and see if I can handle it anywhere. 

P.S.: I have about 80K items that I need to import and right now I have't imported a whole lot of emptying and starting again is okay but if I do finish 50K records and it fails I would be devastated. Please help!!

**Comment by engram-design** [comment]
Out of interest, do you have Dev Mode enabled? Are you receiving any log entries (for either feedme.log or phperrors.log)? Sounds like its an issue of memory usage...

**Comment by subratom** [comment]
I have disabled the dev mode. I don't see any errors after the 500 error code. The files I am uploading to CraftCMS range in 2MB to 20MB. I didn't think that would have been a problem. I have cut off all the items whose main body area might have 1M characters (HTML characters). How much memory is reasonable to run this against? I can run this locally on a 16GB ubuntu box if memory is the issue

**Comment by engram-design** [comment]
If you're getting a 500 error, there should definitely be a `phperrors.log` in your Craft logs folder - can you confirm?

**Comment by subratom** [comment]
It was craft.log and I went through it to locate the last error. I saw it is 503 error. This is was what was found

******************************************************************************************************
2016/06/02 23:24:35 [error] [exception.Craft\HttpException.503] exception 'Craft\HttpException' in /var/www/craftcms/craft/app/etc/web/WebApp.php:934
Stack trace:
#0 /var/www/craftcms/craft/app/etc/web/WebApp.php(242): Craft\WebApp->_processUpdateLogic()
#1 /var/www/craftcms/craft/app/framework/base/CApplication.php(185): Craft\WebApp->processRequest()
#2 /var/www/craftcms/craft/app/index.php(62): CApplication->run()
#3 /var/www/craftcms/public/index.php(19): require_once('/var/www/craftc...')
#4 {main}
REQUEST_URI=/index.php/logon/actions/users/getAuthTimeout?dontExtendSession=1

Sorry I am very new to this CMS so I read and try to make sense of what is going, so please excuse my answers if they don't make sense but I am good at following directions and looking it up. 

Any idea what may be happening? Thank you for your help, I appreciate it.

**Comment by engram-design** [comment]
Hmmm - I wonder if thats an unrelated error? Normally if its a PHP issue, its put in the `phperrors.log` - but it sounds like you're not getting that. Are you running this on a staging/development server? I'd be interested to see your results when testing locally?

**Comment by subratom** [comment]
Yes this is on dev machine. The only way I find to proceed is if I remove the body content of that item, and remove all tasks from tasks table, and restart the process. Would there be a limit on how much data it can take in before it goes belly up?

**Comment by engram-design** [comment]
Yep, there will be a limit - basically whatever amount of memory PHP has available. Its basically got to load your entire feed into memory and deal with that. I suppose 1M characters is a fair bit, combined with other stuff.

One alternative might be to run the feed importing all other fields in first. Then, running again (with Update set as the duplication handling), import just the body content and see if that helps.

Not that it really helps right now, but Feed Me 2.0 will focus more on better performance!

**Comment by subratom** [comment]
Yea I can see that being a problem then. I could definitely try Update method and see if it works. This machine is running in AWS with 4GB RAM and could potentially be a problem with memory running out. I will run this on another machine and see if I get any results.

**Comment by thoughtfulweb** [comment]
In case it helps, I experienced a similar issue when importing a large amount of data using my local MAMP server.

I had upped all the PHP variables, i.e memory_limit and max_execution_time (to silly numbers) but was still getting, what looked like, issues with timeout.

After a lot of head scratching, I discovered there is a limitation with PHP running in Fast CGI mode.

Looking at my Apache log, I discovered this line:

`[Mon May 16 18:59:07 2016] [error] [client ::1] FastCGI: comm with server "/Applications/MAMP/fcgi-bin/php7.0.0.fcgi" **aborted: idle timeout (30 sec**)…`


The idle timeout (30 sec) bit made me curious as I’ve not seen this before, so after searching Stack Exchange, I came across this post: http://craftcms.stackexchange.com/questions/4661/fastcgi-error-when-accessing-a-local-development-site-using-mamp-pro

Essentially I needed to add:

`FastCgiServer /Applications/MAMP/fcgi-bin/**php7.0.0.fcgi** -idle-timeout 3600`

Note that the version of PHP will need to match the version you are running.

I did need use the following info from this post http://stackoverflow.com/questions/24715426/how-do-you-increase-the-apache-fastcgi-timeout-on-mamp-mamp-pro 

> For those that run into this issue I found that simply commenting out the variableMAMP_FastCgiServer_MAMP with a HASH is not acceptable. Only Apache will ignore the HASH. MAMP will still process the file and insert its overrides which will conflict with the ones you are adding. The fix for me was to delete the MAMP_FastCgiServer_MAMPvariable and then add my own overrides. – Hubert Aug 25 '14 at 3:28 

Anyway, the good news  is it stopped the 500 errors I was experiencing. 

I’m not sure if this will work for you, but it may be worth investigating.
